src/algorithms/pathfinders/dijkstra.ts

```diff
@@ -0,0 +1,111 @@
+     * @param mygraph : relevant graph
+     */
+    FindPath(startID: number, endID: number, mygraph: Graph): number[] {
+        this.Initialize(mygraph);
+        this.distances.set(startID, {distance: 0, sourceID: -9001});
+        this.unvisited.delete(startID);
+        let currentVertex = mygraph.getVertex(startID);
+        while (this.unvisited.size > 0) {
+            this.updateDistances(currentVertex);
+            currentVertex = this.getNextVertex(mygraph);
+            if (currentVertex.id === endID) {
+                break;
+            }
+        }
+        return this.producePath(startID, endID);
+    }
+
+    /**
+     * Initializes data structures for finding paths in the given graph
+     * @param mygraph : relevant graph
+     */
+    Initialize(mygraph: Graph): void {
+        this.unvisited = mygraph.getVertexIDs();
+        this.distances = new  Map<number, {
+            distance: number;
+            sourceID: number;
+        }>();
+        for (const vertexID of this.unvisited) {
+            this.distances.set(vertexID, {distance: global.Infinity, sourceID: -9001});
+        }
+    }
+
+    /**
+     * Finds the next best vertex of the graph to visit
+     * @param mygraph : relevant graph
+     */
+    getNextVertex(mygraph: Graph): Vertex {
+        // dumb bruteforce search
+        // TODO: make this something that doesn't suck
+        let bestVertexID: number;
+        let bestDistance: number = global.Infinity;
+        for (const vertexID of this.unvisited) {
+            const entry = this.distances.get(vertexID);
+            if (bestDistance >= entry.distance) {
+                bestDistance = entry.distance;
+                bestVertexID = vertexID;
+            }
+        }
+        return mygraph.getVertex(bestVertexID);
+    }
+
+    /**
+     * Updates the distance map if the distance from given vertex is better than the best known
+     * @param currentVertex : relevant vertex
+     */
+    updateDistances(currentVertex: Vertex): void {
+        this.unvisited.delete(currentVertex.id);
+        const currentDistance = this.distances.get(currentVertex.id).distance;
+        for (const neighbor of currentVertex.neighbors) {
+            if (!this.unvisited.has(neighbor.vertex.id)) {
+                continue;
+            }
+            const newDistance = currentDistance + neighbor.costs.distance;
+            const oldDistanceEntry = this.distances.get(neighbor.vertex.id);
+            if (oldDistanceEntry === undefined || oldDistanceEntry.distance > newDistance) {
+                this.distances.set(neighbor.vertex.id, {distance: newDistance, sourceID: currentVertex.id});
+            }
+        }
+    }
+
+    /**
+     * Produces an array representing the path from startID to endID based on the distance map
+     * @param startID : ID of starting vertex
+     * @param endID : ID of destination vertex
+     */
+    producePath(startID: number, endID: number): number[] {

```

Comment:
All those methods could be private as you call all of them from the FindPath() method (that should be the only one public). But as I suggested above, I'd rather see all those methods in the Graph class itself.